src/runtime/Symbol.cpp

```diff
@@ -26,14 +26,14 @@ namespace Escargot {
 
 size_t g_symbolTag;
 
-Symbol* Symbol::fromGlobalSymbolRegistry(VMInstance* vm, String* stringKey)
+Symbol* Symbol::fromGlobalSymbolRegistry(VMInstance* vm, Optional<String*> stringKey)
 {
     // Let stringKey be ? ToString(key).
     // For each element e of the GlobalSymbolRegistry List,
     auto& list = vm->globalSymbolRegistry();
     for (size_t i = 0; i < list.size(); i++) {
         // If SameValue(e.[[Key]], stringKey) is true, return e.[[Symbol]].
-        if (list[i].key->equals(stringKey)) {
+        if (list[i].key->equals(stringKey.value())) {
```

Comment:
What if user gives stringKey as empty?

Reply:
IMO Symbol.for can not accept stringKey as empty.
thus we don't have to use Optinal<String*> for stringKey